DisguiseUnityRenderStream/Runtime/Util/MulticastExtensions.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.NetworkInformation;
+using System.Net.Sockets;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Disguise.RenderStream.Utils
+{
+    static class MulticastExtensions
+    {
+        public static void EnableMulticast(this UdpClient udpClient, IPAddress multicastIPAddress, int port, string adapterName = null, int timeoutMs = 5000)
+        {
+            var (selectedNic, selectedNicAddress) = SelectNetworkInterface(adapterName);
+            if (selectedNic == null)
+            {
+                throw new IOException("There are no available network interfaces that support multicast");
+            }
+
+            Assert.IsNotNull(selectedNicAddress);
+            var adapterAddress = selectedNicAddress.Address;
+
+            udpClient.Client.SendTimeout = timeoutMs;
+            udpClient.Client.ReceiveTimeout = timeoutMs;
+
+            // Ask to send the multicast traffic using the specified adapter
+            udpClient.Client.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastInterface,
+                adapterAddress.GetAddressBytes());
+
+            // There should be only one hop between the emitter and repeater
+            // Food for thought: Should we make this configurable to work on slightly more complex network infrastructures?
+            udpClient.Client.SetSocketOption(SocketOptionLevel.IP, SocketOptionName.MulticastTimeToLive, 1);
+
+            // Allow multiple ClusterDisplay applications to bind on the same address and port. Useful for when running
+            // multiple nodes locally and unit testing.
+            // Food for thought: Does it have a performance cost? Do we want to have it configurable or disabled in some
```

Comment:
Opinion: I don't think performance matter that much for this part of the code, this is only a negotiation when starting...